docs/recommendations.rst

```diff
@@ -0,0 +1,81 @@
+Recommendations
+===============
+
+Recommendations offer a quick shortcut for helping tune your policy configurations in an environment, by providing
+suggested reputation overrides which you may add to improve your policy. They can speed up the process of tuning your
+policy to an environment, rather than having to manually investigate endpoint activity and reconfigure the policy in
+response to those investigations.
+
+The Carbon Black Cloud SDK for Python offers assistance for dealing with Recommendations.
+
+Getting the List of Recommendations
+-----------------------------------
+
+By querying the ``Recommendation`` object, you can see which recommendations have already been generated for you by
+the Carbon Black Cloud.
+
+::
+
+    >>> from cbc_sdk import CBCloudAPI
+    >>> api = CBCloudAPI(profile='sample')
+    >>> from cbc_sdk.endpoint_standard import Recommendation
+    >>> query = api.select(Recommendation).set_statuses(['NEW', 'ACCEPTED', 'REJECTED']).sort_by('impact_score', 'DESC')
+    >>> recslist = list(query)
+    >>> for rec in recslist:
+    ...     print(f"{rec.recommendation_id} - {rec.workflow_.status} - {rec.impact_.impact_score}")
+    ...
+    5e6926d4-0c55-4757-a94d-e05883d5ee4c - ACCEPTED - 1.1710311
```

Comment:
I don't know if this is possible, but for an API consumer it would be far superior to display the recommendation text than the recommendation ID. The ID is necessary for performing operations, but I can't make an effective decision which to leverage if I don't know what it's recommending.

Unfortunately because I can't find any recommendations in any of my test orgs I'm not sure what exactly to suggest here.

Reply:
This is just an _example._ It is not intended to teach the reader _everything_ about dealing with recommendations, just show how the API accesses them.

Reply:
I appreciate that examples can't spell out everything, but on the assumption that some users will copy our code and run it as-is - and to illustrate something more recognizable to the user (make the connection to the thing we're pulling out with this script - nearly any other attribute would be more relevant/illustrative than the random identifier value. If it'd help for me to suggest something, I'd love to help - but I don't know of an org where there are Recommendations *and* where I can generate an API key.

Reply:
I have revised this example to include also the filename and override list fields from the new rule.  This is the best I can do for you; it's _hard_ finding a combination of fields that will convey useful information, given how many of the Recommendations in our dev organization have "junk" values for some of those fields (like `None`). It took me four tries before I came up with _this_ combination, and, even then, I had to cut a _lot_ of the returned records because they had `None` in one of the columns, like the filename one.

Reply:
What does it look like when you do `print(rec)`? Given the subclass structure of the object does it print all of the necessary fields? Does this class need a custom `__str__` method?

Reply:
That is what we call a Darn Good Question. Fortunately, I have an answer:
[output_list.txt](https://github.com/carbonblack/carbon-black-cloud-sdk-python/files/7237964/output_list.txt)

As you can see, this output is both (a) truncated in the case of the subobjects, and (b) _way_ too long to include as sample output in the middle of a guide doc, even if I edited it down.

Reply:
I would consider overriding the `__str__` method for Recommendation and implement something that does a better job of printing that. Similar to what we need to `Process.Tree` and `Process.Summary`

Reply:
It's a little late in the game to go back and change the code, isn't it?

Reply:
I'm not asking for something to change but rather to make an additive change to include a `__str__` function in the Recommendation class to better print that what is currently being exposed

Reply:
On the inline example front you nailed it @abowersox-cb - thank you for pawing through the data to show something that's more immediately recognizable as the kinds of outcomes the API allows.

I can see the value of such a `__str__` function, and would gladly support that if it could accompany this guidance.

Reply:
Is there a way we can just declare what we have as Good Enough For Now, finish this ticket, and then create another one to deal with the `__str__` issue? Because putting that in seems to be going _way_ out of scope for this particular ticket.

Reply:
Sure. As long as it is finished before the release